### PR TITLE
Check pointers against nullptr.

### PR DESCRIPTION
**Pull request by hbina**

### CODE REVIEW

**Comment by akien-mga** [comment]
Why?

**Comment by hbina** [comment]
So something like this will not compile. Makes for less confusing debugging experience in the future.
https://godbolt.org/z/_jWspm

**Comment by Calinou** [comment]
As I discussed with @akien-mga a few days ago, we considered removing the `ERR_FAIL_NULL` macros entirely in favor of using `ERR_FAIL_COND(something == nullptr)` instead. It would be less confusing for newer contributors, since there would be less error checking macros to choose from.

That said, I think this PR is still a good change. It prevents parameters that evaluate to `false` from throwing an error.

File: core/error_macros.h [hunk 1]
```diff
@@ -285,7 +285,7 @@ void _err_print_index_error(const char *p_function, const char *p_file, int p_li
  * If it is null, the current function returns.
  */
 #define ERR_FAIL_NULL(m_param)                                                                          \
-	if (unlikely(!m_param)) {                                                                           \
+	if (unlikely(m_param == nullptr)) {                                                                           \
```
Comment (akien-mga) [inline core/error_macros.h:288]: You need to fix the alignment of the trailing `\`: https://travis-ci.org/github/godotengine/godot/jobs/697742242

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**